sql/driver/doc.go

```diff
@@ -0,0 +1,284 @@
+
+	q :=`SELECT name, age FROM person WHERE age >= ?`
+	rows, err := db.Query(q, 30)
+	// handle the error
+	defer rows.Close()
+	var name string
+	var age int
+	for rows.Next() {
+		err := rows.Scan(&name, &age)
+		// handle the error
+		fmt.Println(name, age)
+	}
+
+Context variants of Query and Exec, such as QueryContext and ExecContext are fully supported.
+Context variants are used to pass Hazelcast specific parameters.
+
+Passing Hazelcast-Specific Parameters
+
+This driver supports the following extra query parameters that Hazelcast supports:
+
+	- Cursor buffer size: Size of the server-side buffer for rows.
+	- Timeout: Maximum time a query is allowed to execute.
+
+Checkout the documentation below for details.
+
+The extra query parameters are passed in a context augmented using WithCursorBufferSize and WithQueryTimeout functions. Here is an example:
+
+	// set the cursor buffer size to 10_000
+	ctx := driver.WithCursorBufferSize(context.Background(), 10_000)
+	// set the query timeout to 2 minutes
+	ctx = driver.WithQueryTimeout(ctx, 2*time.Minute)
+	// use the parameters above with any methods that uses that context
+	rows, err := db.QueryContext(ctx, "select * from people")
+
+Creating a Mapping
+
+To connect to a data source and query it as if it is a table, a mappings should be created.
+Currently, mappings for Map, Kafka and file data sources are supported.
+
+You can read the details about mappings here: https://docs.hazelcast.com/hazelcast/latest/sql/sql-overview#mappings
+
+Supported Data Types
+
+The following data types are supported, as both builtin values and in types supporting the portable interface when inserting/updating:
+
+	- string (varchar)
+	- int8 (tinyint)
+	- int16 (smallint)
+	- int32 (integer)
+	- int64 (bigint)
+	- bool (boolean)
+	- float32 (real)
+	- float64 (double)
+	- types.Decimal (decimal)
+	- time.Time (date) Detected by checking: hour == minute == second == nanoseconds = 0
+	- time.Time (timestamp) Detected by checking: year == 0, month == day == 1, timezone == time.Local
+	- time.Time (timestamp with time zone) Detected by checking: year == 0, month == day == 1, timezone != time.Local
+
+Using Raw Values
+
+You can directly use one of the supported data types.
+
+Creating a mapping:
+
+        CREATE MAPPING person
+        TYPE IMAP
+        OPTIONS (
+            'keyFormat' = 'int',
+            'valueFormat' = 'varchar'
+        )
+
+Inserting rows:
+
+	INSERT INTO person VALUES(100, 'Jane Doe')
+
+Querying rows:
+
+	SELECT __key, this from person
+
+Using JSON
+
+Non-nested JSON values are supported.
+
+Assuming the following JSON value:
+
+	{
+		"age": 35,
+		"name": "Jane Doe"
+	}
+
+Some or all fields of the JSON value may be mapped and used.
+
+Creating a mapping:
+
+        CREATE MAPPING person (
+			__key BIGINT,
+			age BIGINT,
+			name VARCHAR
+		)
+        TYPE IMAP
+        OPTIONS (
+            'keyFormat' = 'bigint',
+            'valueFormat' = 'json-flat'
+        )
```

Comment:
I see this code sample as indented, I don't know why. 
![image](https://user-images.githubusercontent.com/32355782/147761293-76bf2b79-8114-4e35-807a-794065fa8aee.png)


all other code samples are rendered normally.

Reply:
My bad, extra indentation.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/709/commits/b0798c9d652c2a57683c730b872d5d2fc19d71c0